src/vs/workbench/contrib/accessibilitySignals/browser/editorTextPropertySignalsContribution.ts

```diff
@@ -164,20 +166,9 @@ export class EditorTextPropertySignalsContribution extends Disposable implements
 	}
 
 	private _getDelay(signal: AccessibilitySignal, modality: AccessibilityModality): number {
-		// TODO make these delays configurable!
-		if (signal === AccessibilitySignal.errorAtPosition || signal === AccessibilitySignal.warningAtPosition) {
-			if (modality === 'sound') {
-				return 100;
-			} else {
-				return 1000;
-			}
-		}
-
-		if (modality === 'sound') {
-			return 400;
-		} else {
-			return 3000;
-		}
+		const delaySettingsKey = signal.delaySettingsKey ?? 'accessibility.signalOptions.delays.general';
+		const delaySettingsValue: { sound: number; announcement: number } = this._configurationService.getValue(delaySettingsKey);
+		return modality === 'sound' ? delaySettingsValue.sound : delaySettingsValue.announcement;
```

Comment:
This logic should be in `IAccessibilitySignalService`.
I think the entire function can be moved there (`getDelayMs`).